media/transcoder/create_job_with_concatenated_inputs.py

```diff
@@ -0,0 +1,202 @@
+# limitations under the License.
+
+"""Google Cloud Transcoder sample for creating a job based on concatenating two input videos.
+
+Example usage:
+    python create_job_with_concatenated_inputs.py --project_id <project-id> --location <location> \
+      --input1_uri <uri> --start_time_input1 <sec> --end_time_input1 <sec> \
+      --input2_uri <uri> --start_time_input2 <sec> --end_time_input2 <sec> \
+      --output_uri <uri>
+"""
+
+# [START transcoder_create_job_with_concatenated_inputs]
+
+import argparse
+
+from google.cloud.video import transcoder_v1
+from google.cloud.video.transcoder_v1.services.transcoder_service import (
+    TranscoderServiceClient,
+)
+from google.protobuf import duration_pb2 as duration
+
+
+def create_job_with_concatenated_inputs(
+    project_id,
+    location,
+    input1_uri,
+    start_time_input1,
+    end_time_input1,
+    input2_uri,
+    start_time_input2,
+    end_time_input2,
+    output_uri,
+):
+    """Creates a job based on an ad-hoc job configuration that concatenates two input videos.
+
+    Args:
+        project_id: The GCP project ID.
+        location: The location to start the job in.
+        input1_uri: Uri of the first video in the Cloud Storage bucket.
+        start_time_input1: Start time, in fractional seconds ending in 's'
+          (e.g., '0s'), relative to the first input video timeline.
+        end_time_input1: End time, in fractional seconds ending in 's'
+          (e.g., '8.1s'), relative to the first input video timeline.
+        input2_uri: Uri of the second video in the Cloud Storage bucket.
+        start_time_input2: Start time, in fractional seconds ending in 's'
+          (e.g., '3.5s'), relative to the second input video timeline.
+        end_time_input2: End time, in fractional seconds ending in 's'
+          (e.g., '15s'), relative to the second input video timeline.
+        output_uri: Uri of the video output folder in the Cloud Storage bucket."""
+
+    s1 = duration.Duration()
+    s1.FromJsonString(start_time_input1)
+    e1 = duration.Duration()
+    e1.FromJsonString(end_time_input1)
+
+    s2 = duration.Duration()
+    s2.FromJsonString(start_time_input2)
+    e2 = duration.Duration()
+    e2.FromJsonString(end_time_input2)
+
+    client = TranscoderServiceClient()
+
+    parent = f"projects/{project_id}/locations/{location}"
+    job = transcoder_v1.types.Job()
+    job.output_uri = output_uri
+    job.config = transcoder_v1.types.JobConfig(
+        inputs=[
+            transcoder_v1.types.Input(
+                key="input1",
+                uri=input1_uri,
+            ),
+            transcoder_v1.types.Input(
+                key="input2",
+                uri=input2_uri,
+            ),
+        ],
+        edit_list=[
+            transcoder_v1.types.EditAtom(
+                key="atom1",
+                inputs=["input1"],
+                start_time_offset=s1,
+                end_time_offset=e1,
```

Comment:
proto-plus can also take the Python `datetime.timedelta`type. https://proto-plus-python.readthedocs.io/en/stable/marshal.html#well-known-types

Instantiating a protobuf Duration is fine as well, I just wanted to offer options since I know this isn't documented very well right now. 😄 

```py
>>> import datetime
>>> from google.cloud.video import transcoder_v1
>>> a = transcoder_v1.EditAtom()
>>> a.end_time_offset=datetime.timedelta(seconds=3)
>>> a
end_time_offset {
  seconds: 3
}
>>> a.end_time_offset=datetime.timedelta(seconds=3.5)
>>> a
end_time_offset {
  seconds: 3
  nanos: 500000000
}

Reply:
Thank you! I actually did some research trying to find something like this but couldn't. I think I will keep Duration since the other samples do also, but will use this in the future.